ci/do.sh

```diff
@@ -0,0 +1,159 @@
+    _PRETTY_NAME=$(echo ${PRETTY_NAME} | awk -F[=\"] '{print $1}')
+    _PRETTY_NAME=${_PRETTY_NAME# }
+    _PRETTY_NAME=${_PRETTY_NAME% }
+    _PRETTY_NAME=${_PRETTY_NAME// /_}
+    _base_link="http://download.opensuse.org/repositories"
+    zypper -n ar -f -G ${_base_link}/devel:/tools/${_PRETTY_NAME}/devel:tools.repo
+    zypper -n ar -f -G ${_base_link}/devel:/libraries:/c_c++/${_PRETTY_NAME}/devel:libraries:c_c++.repo
+    zypper -n ref
+    zypper -n update
+    zypper -n install rpmdevtools python3-pip sudo which net-tools man time.x86_64 git
+    rpmdev-setuptree
+    zypper -n install --force-resolution $(rpmspec --buildrequires -q ${SPEC_FILE} | sort -u | grep -vE '^(/bin/)?(ba)?sh$')
+    zypper -n install --force-resolution $(rpmspec --requires -q ${SPEC_FILE} | sort -u | grep -vE '^(/bin/)?(ba)?sh$')
+    pip3 install -r ${REQ_FILE}
+  elif [ "x${ID}" == "xdebian" ]; then
+    if [ "x${DEBIAN_FRONTEND}" == "x" ]; then
+      export DEBIAN_FRONTEND=noninteractive
+    fi
+    apt-get -y update
+    apt-get -y upgrade
+    apt-get install -y build-essential dpkg-dev autoconf libtool rpm alien libssl-dev \
+                        libxt-dev libpq-dev libexpat1-dev libedit-dev libncurses5-dev \
+                        libical-dev libhwloc-dev pkg-config tcl-dev tk-dev python3-dev \
+                        swig expat postgresql postgresql-contrib python3-pip sudo man-db git
+    pip3 install -r ${REQ_FILE}
+  elif [ "x${ID}" == "xubuntu" ]; then
+    if [ "x${DEBIAN_FRONTEND}" == "x" ]; then
+      export DEBIAN_FRONTEND=noninteractive
+    fi
+    apt-get -y update
+    apt-get -y upgrade
+    apt-get install -y build-essential dpkg-dev autoconf libtool rpm alien libssl-dev \
+                        libxt-dev libpq-dev libexpat1-dev libedit-dev libncurses5-dev \
+                        libical-dev libhwloc-dev pkg-config tcl-dev tk-dev python3-dev \
+                        swig expat postgresql python3-pip sudo man-db git
+    pip3 install -r ${REQ_FILE}
+  else
+    echo "Unknown platform..."
+    exit 1
+  fi
+fi
+
+if [ "x${FIRST_TIME_BUILD}" == "x1" -a "x${IS_CI_BUILD}" == "x1" ]; then
+  echo "### First time build is complete ###"
+  exit 0
+fi
+
+if [ "x${ONLY_INSTALL_DEPS}" == "x1" ]; then
+  exit 0
+fi
+
+_targetdirname=target-${ID}
+if [ "x${ONLY_INSTALL}" != "x1" -a "x${ONLY_REBUILD}" != "x1" -a "x${ONLY_TEST}" != "x1" ]; then
+  rm -rf ${_targetdirname}
+fi
+mkdir -p ${_targetdirname}
+if [ "x${ONLY_REBUILD}" != "x1" -a "x${ONLY_INSTALL}" != "x1" -a "x${ONLY_TEST}" != "x1" ]; then
+  [[ -f Makefile ]] && make distclean || true
+  ./autogen.sh
+  _cflags="-g -O2 -Wall -Werror"
+  if [ "x${ID}" == "xubuntu" ]; then
+    _cflags="${_cflags} -Wno-unused-result"
+  fi
+  cd ${_targetdirname}
+  if [ -f /src/ci ];then
+    if [ -f ${workdir}/.configure_opt ]; then
+      configure_opt="$(cat ${workdir}/.configure_opt)"
+      _cflags="$(echo ${configure_opt} | awk -F'"' '{print $2}')"
```

Comment:
I guess you mean configure_opt instead _cflags here right? else it will override _cflags from line no. 99 and 101

Reply:
It won't override the _cflags unless the user has specified new cflags. The 2 conditionals check wether it's a ci build and wether .configure_opt file is created by the user. This will ensure it only overrides default when user manually specifies.